File: smart-parking-app/parking-availability-service/src/main/java/com/adb/model/Spot.java
package com.adb.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Reference;

import java.io.Serializable;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Spot implements Serializable{
    @Id
    private Long spotId ;
    private  int number;
    private float price;
    private Boolean availabilityStatus;
    private String vehicleType;
    private Boolean disabled ;
    @Reference
    private Parking parking;


}

File: smart_parking_front_end/lib/models/Spot.dart
import 'Parking.dart';

class Spot {
  final int spotId;
  final int number;
  final double price;
  final bool availabilityStatus;
  final String vehicleType;
  final bool disabled;
  final Parking parking; // Add this field

  Spot({
    required this.spotId,
    required this.number,
    required this.price,
    required this.availabilityStatus,
    required this.vehicleType,
    required this.disabled,
    required this.parking, // Add this required parameter
  });

  factory Spot.fromJson(Map<String, dynamic> json) {
    return Spot(
      spotId: json['spotId'],
      number: json['number'],
      price: json['price'],
      availabilityStatus: json['availabilityStatus'],
      disabled: json['disabled'],
      vehicleType: json['vehicleType'],
      parking:
          Parking.fromJson(json['parking']), // Create Parking object from JSON
    );
  }
}
